ni_measurementlink_service/session_management/_reservation.py

```diff
@@ -1305,7 +1319,10 @@ def create_niscope_session(
             options,
             initialization_behavior,
         )
-        return self._create_session_core(session_constructor, INSTRUMENT_TYPE_NI_SCOPE)
+        closing_function = functools.partial(closing_session, initialization_behavior)
```

Comment:
In this PR, only the NI-Scope specific APIs are updated i.e., the `create_niscope_session` and `create_niscope_sessions` driver APIs in _reservation.py and the `create_niscope_sessions` and `destroy_niscope_sessions` in the [teststand_niscope.py](https://github.com/ni/measurementlink-python/blob/users/jay/enum_support_for_detach_close/examples/niscope_acquire_waveform/teststand_niscope.py). 

All the other driver API and TestStand code module updates will be done as part of [AB#2552333](https://dev.azure.com/ni/DevCentral/_workitems/edit/2552333).